Content.Server/GameObjects/Components/Stack/StackComponent.cs

```diff
@@ -23,9 +24,34 @@ public class StackComponent : SharedStackComponent, IInteractUsing, IExamine
         public override int Count
         {
             get => base.Count;
-            set => base.Count = value;
+            set
+            {
+                base.Count = value;
+                
+                if (base.Count != 0 && SpriteStates != null && Owner.TryGetComponent<SpriteComponent>(out var spriteComponent))
+                {
+                    if(Count == MaxCount)
+                    {
+                        spriteComponent.LayerSetState(0, SpriteStates[2]);
```

Comment:
Indexing these is probably bad and should use steps like the other sprites do (e.g. tasers).

Should probably use an enum for the layer but I guess using 0's fine for now.

Also I feel like this won't work if netsync is disabled for the sprite (which is a lot of them to minimise bandwidth) so it needs to be handled client-side.

I'd just use the client's count and update an AppearanceComponent whenever it changes (might need to make a separate visualizer for it).

Reply:
I thought about being able to pick single items out of a stack lying around, would that still work client-side? Should the client then just send an update to the server?

Reply:
Interactions aren't predicted yet so no. Currently the client's count is updated when the server sends a new state update so whenever that happens the appearance should be updated (this is what HandleComponentState does).